docs/reference/rate-limit-external.md

```diff
@@ -0,0 +1,25 @@
+# Rate Limiting with an External Rate Limit Service
+
+Ambassador can validate incoming requests before routing them to a backing service. When using an external rate limit service, Ambassador will generate a gRPC request to the external rate limit service and will provide a list of descriptors on which the rate limit service can base it's decision to accept or reject the request:
+
+```
+[
+  {"source_cluster", "<local service cluster>"},
+  {"destination_cluster", "<routed target cluster>"},
+  {"remote_address", "<trusted address from x-forwarded-for>"},
+  {"generic_key", "<descriptor_value>"},
+  {"<some_request_header>", "<header_value_queried_from_header>"}
+]
+```
+
+This gRPC service must implement the Envoy [ratelimit.proto](https://github.com/envoyproxy/envoy/blob/master/source/common/ratelimit/ratelimit.proto).
+
+If Ambassador cannot contact the rate limit service, it will allow the request to be processed as if there were no rate limit service configuration.
+
+If Ambassador receives an `OK` response from the rate limit service, then Ambassador allows the client request to resume being processed by the normal Ambassador Envoy flow. If Ambassador receives an `OVER_LIMIT` response, then Ambassador will return an HTTP 429 response to the client.
+
```

Comment:
I feel like you need to talk a bit about the actual rate limit returned from the rate-limiting service -- e.g. the sample service you provide responds `OK` with a 1000 rps limit. Talking briefly about what  Envoy actually does with that part of the response will help people understand the overall rate-limiting system.